### DISCUSSION TITLE
Incomplete diagnostics

### DISCUSSION THREAD

**Discussion by spadova-a** [BLAST-WarpX/warpx#4488] [Q&A]
Hello there,
I'm trying to run a simulation using the boosted frame technique. I started with the example provided on your website 
[inputs_2d_boost (1).txt](https://github.com/ECP-WarpX/WarpX/files/13602744/inputs_2d_boost.1.txt) and changed the output format for the diagnostics to .h5. The simulation runs ok, but when I increase the `amr.max_grid_size` I can only see a part of the diagnostics - see the picture below.
![image](https://github.com/ECP-WarpX/WarpX/assets/88655409/fb8bcacf-77c1-4f2b-add7-cb64134f9a9d)
Do you have any idea why is this happening? I want to do simulations with bigger grid and higher resolution on GPU, so I will have to use `amr.max_grid_size > 64` for the simulation to be effective.
Thank you in advance.

**Comment by RemiLehe** [comment]
Hi,
I think that what you are seeing suggests that the simulation was not run for enough timesteps.
Did you get a warning message in the WarpX text output that says that the back-transformed diagnostics are not complete, and that the simulation should be run longer?

**Comment by spadova-a** [comment]
Hi, yes thank you, that solved the problem. However, is there something like minimum value for `diag1.dt_snapshots_lab`? Because I tried to set it to `diag1.dt_snapshots_lab = 1.2e-12` and do 4 snapshots (`diag1.num_snapshots_lab = 4`). The total time of the simulation was 5e-12 but I got this message and incomplete picture from the diagnostic again...

![image](https://github.com/ECP-WarpX/WarpX/assets/88655409/87aaf032-a9a9-4047-ad89-fbc69e35cc52)

Also, how can the number of snapshots influence the diagnostics? I would expect, that if I set too many snaphostsI would just get only a part of them... Is this somehow related to the way this BTD is implemented?

**Comment by RemiLehe** [comment]
Yes, that is correct, if you have too many snapshots, the last few snapshots will be incomplete (or may not even be created in some cases), unless you run the simulation for long enough.
Here is some more info: https://warpx.readthedocs.io/en/latest/usage/faq.html#what-about-back-transformed-diagnostics-btd

**Comment by spadova-a** [comment]
Great, it is much more clear to me now. Thanks for your help, I will close the discussion.
